src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -721,9 +721,12 @@ public SyntaxNode GetContainingMemberDeclaration(SyntaxNode root, int position)
             var node = root.FindToken(position).Parent;
             while (node != null)
             {
-                if (node is MemberDeclarationSyntax)
+                if (useFullSpan || node.Span.Contains(position))
                 {
-                    return node;
+                    if (node is MemberDeclarationSyntax)
```

Comment:
Why the change here? What about compound things like accessors or something? What should be returned there?